File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n=nums.size();
        int left=0;
        int sum=0;
        int mini=INT_MAX;
        for(int i=0;i<n;i++){
            sum+=nums[i];
            while(sum>=target){
                mini=min(mini,i-left+1);
                sum-=nums[left];
                left++;
            }
        }
        if(mini==INT_MAX){
            return 0;
        }
        return mini;
    
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        left=0
        sum=0
        minn=n+1
        for r in range(n):
            sum+=nums[r]
            while(sum>=target):
                minn=min(minn,r-left+1)
                sum-=nums[left]
                left+=1
        if minn==n+1:
            return 0
        return minn
        